src/main/sensors/temperature.c

```diff
@@ -25,27 +25,36 @@
 #include "config/parameter_group.h"
 #include "config/parameter_group_ids.h"
 
+#include "fc/runtime_config.h"
+
+#include "sensors/sensors.h"
 #include "sensors/temperature.h"
 #include "sensors/gyro.h"
+#include "sensors/barometer.h"
 
-static bool     tempSensorValid[TEMP_COUNT];
 static int16_t  tempSensorValue[TEMP_COUNT];
+static tempSensor_e tempSensorValid;
 
-bool isTemperatureSensorValid(tempSensor_e sensor)
+int16_t getTemperature(tempSensor_e sensor)
 {
-    return tempSensorValid[sensor];
+    return tempSensorValue[sensor];
 }
 
-int16_t getTemperature(tempSensor_e sensor)
+int16_t getCurrentTemperature(void)
 {
-    return tempSensorValue[sensor];
+    return tempSensorValue[tempSensorValid];
 }
 
 void temperatureUpdate(void)
 {
-    // TEMP_GYRO: Update gyro temperature
+    // TEMP_GYRO: Update gyro temperature in decidegrees
     if (gyroReadTemperature()) {
-        tempSensorValid[TEMP_GYRO] = true;
         tempSensorValue[TEMP_GYRO] = gyroGetTemperature();
+        tempSensorValid=TEMP_GYRO;
+    }
+    // TEMP_BARO: Update baro temperature in decidegrees
+    if(sensors(SENSOR_BARO)){
+        tempSensorValue[TEMP_BARO] = baroGetTemperature();
+        tempSensorValid=TEMP_BARO;
```

Comment:
If baro and gyro will read different temperatures - output of `getCurrentTemperature` will jump.